src/canvas/canvas_events.ts

```diff
@@ -856,14 +856,20 @@ export class Canvas extends SelectableCanvas {
         originalMouseUpHandler(e, transform, pointer.x, pointer.y);
     }
     this._setCursorFromEvent(e, target);
-    this._handleEvent(e, 'up', LEFT_CLICK, isClick);
+    this._handleEvent(e, 'up', LEFT_CLICK, this._isClick);
     this._groupSelector = null;
     this._currentTransform = null;
     // reset the target information about which corner is selected
     target && (target.__corner = 0);
     if (shouldRender) {
       this.requestRenderAll();
-    } else if (!isClick) {
+    } else if (
+      !this._isClick &&
+      !(
+        isInteractiveTextObject(this._activeObject) &&
+        this._activeObject.isEditing
+      )
+    ) {
```

Comment:
this wasn't necessary before, what kind of bug is causing?
Is because the renderTop is not necessary or because the renderTop was causing a visual artifact?

Reply:
The bug was in `isClick`, it was wrong so logic was executing correctly...
What happens is top context renders and clears text selection.
I thought instead of this check to render text selection for `renderTop`, making it more robust
Thoughts?